src/Http/WebUtilities/src/HttpResponseStreamWriter.cs

```diff
@@ -210,7 +210,7 @@ public override Task WriteAsync(char[] values, int index, int count)
         private async Task WriteAsyncAwaited(char[] values, int index, int count)
         {
             Debug.Assert(count > 0);
-            Debug.Assert(_charBufferSize - _charBufferCount > count);
+            Debug.Assert(_charBufferSize - _charBufferCount < count);
```

Comment:
This method is called specifically when there isn't enough room in the buffer. See 
https://github.com/aspnet/AspNetCore/pull/6307/files#diff-6058ce1750996ffd08f179e654b5e9b9R198
https://github.com/aspnet/AspNetCore/pull/6307/files#diff-6058ce1750996ffd08f179e654b5e9b9R259